### PR TITLE
Adding database model.

### PR DESCRIPTION
**Pull request by n3en**

### CODE REVIEW

**Review by Guillaum6e — APPROVED** [review]

File: database.sql [hunk 1]
```diff
@@ -1,63 +1,43 @@
...
-  MODIFY `id` int(11) UNSIGNED NOT NULL AUTO_INCREMENT, AUTO_INCREMENT=3;
-/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
-/*!40101 SET CHARACTER_SET_RESULTS=@OLD_CHARACTER_SET_RESULTS */;
-/*!40101 SET COLLATION_CONNECTION=@OLD_COLLATION_CONNECTION */;
+CREATE TABLE user (
+  id INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+  pseudo VARCHAR(20) NOT NULL,
+  adress VARCHAR(80) NOT NULL,
+  email VARCHAR(50) NOT NULL,
+  photo LONGBLOB NOT NULL,
```
Comment (Benja-san) [inline database.sql]: No pictures in database, the best practice would be to import pictures on the public folder then store the url only (as a text field) on database.

File: database.sql [hunk 2]
```diff
@@ -1,63 +1,43 @@
...
+  rating INT DEFAULT 5
+);
+
+
+CREATE TABLE product (
+  id INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+  title VARCHAR(20) NOT NULL,
+  price INT NOT NULL,
+  description TEXT NOT NULL,
+  photo LONGBLOB NOT NULL,
```
Comment (Benja-san) [inline database.sql]: See the first comment aout pictures

File: database.sql [hunk 3]
```diff
@@ -1,63 +1,43 @@
...
+  `date` DATETIME DEFAULT NOW(),
+  user_id INT NOT NULL,
+  CONSTRAINT fk_cart_user
+        FOREIGN KEY (user_id)
+        REFERENCES user(id),
+  product_id INT NOT NULL,
+  CONSTRAINT fk_cart_product
+        FOREIGN KEY (product_id)
+        REFERENCES product(id)
+);
```
Comment (Benja-san) [inline database.sql:48]: Don't forget to add the On DELETE and ON UPDATE options in order to delete all necessaries data when tables have relations

**Review by Benja-san — CHANGES_REQUESTED** [review]
Changes requested for picture columns

File: database.sql (hunk 1 above)
Comment (Markus0059) [reply database.sql]: Nicolas changed the type to varchar type

File: database.sql [hunk 4]
```diff
@@ -1,63 +1,49 @@
...
+  CONSTRAINT fk_cart_user
+        FOREIGN KEY (user_id)
+        REFERENCES user(id)
+        ON DELETE CASCADE
+        ON UPDATE NO ACTION,
+  product_id INT NOT NULL,
+  CONSTRAINT fk_cart_product
+        FOREIGN KEY (product_id)
+        REFERENCES product(id)
+        ON DELETE CASCADE
```
Comment (Benja-san) [inline database.sql]: no action would be a better solution

File: database.sql [hunk 5]
```diff
@@ -1,63 +1,49 @@
...
+  id INT PRIMARY KEY AUTO_INCREMENT NOT NULL,
+  status_validation BOOL DEFAULT false,
+  `date` DATETIME DEFAULT NOW(),
+  user_id INT NOT NULL,
+  CONSTRAINT fk_cart_user
+        FOREIGN KEY (user_id)
+        REFERENCES user(id)
+        ON DELETE CASCADE
+        ON UPDATE NO ACTION,
+  product_id INT NOT NULL,
```
Comment (Benja-san) [inline database.sql]: Here you won't really need a product id, but instead your product might need a cart_id cause a cart can handle 0 or n products

**Review by Benja-san — CHANGES_REQUESTED** [review]

**Review by Benja-san — APPROVED** [review]

**Pull request merged**